foundry.toml

```diff
@@ -17,6 +17,32 @@ solc = "0.8.23"
 # IPFS hash to the bytecode.
 bytecode_hash = "none"
 ffi = true
+fs_permissions = [{ access = "read-write", path = "./contracts/script/"}]
+
+[profile.defender]
+# Note that if we want to move the `./contracts` directory into it's own repo we
+# need to move the foundry.toml and remove the `./contracts` prefix from the
+# directories below.
+src = 'contracts/src'
+out = 'out'
+test = 'contracts/test'
+libs = ['contracts/lib']
+script = "contracts/script"
+broadcast = "contracts/broadcast"
+cache_path = "contracts/cache"
+# Version should match the solc installed via flake, otherwise the contract
+# artifacts may differ slightly.
+solc = "0.8.23"
+# The bytecode hash is no longer deterministic and prevents us from generating
+# deterministic bindings. The "none" setting prevents solc from appending the
+# IPFS hash to the bytecode.
+bytecode_hash = "none"
+ffi = true
+# needed for openzeppelin defender
+build_info = true
```

Comment:
So my suggestion is to add `extra_output` and `fs_permissions` to the default profile and then run `forge script` with `--build-info` when we do a defender deployment. We can also keep the `defender` profile. Note that it should not be necessary to duplicate the entire `default` profile because all other profiles inherit values from the default profile.

Reply:
I've removed duplications in the defender profile: https://github.com/EspressoSystems/espresso-sequencer/commit/d17dac1a2bffb30144de9584a30cf39bcd2668ea. 

Using --build-info true in the script command works but defender also requires the files to be built in `out` instead of `contracts/out` which is the current setting in the default profile. 

Do you have a preference for where the build files go? the `gen-bindings` currently uses contracts/out